packages/dht/src/connection/ConnectionManager.ts

```diff
@@ -76,11 +77,20 @@ export interface TlsCertificate {
     certFileName: string
 }
 
-interface Endpoint {
-    connection: ManagedConnection
+interface ConnectingEndpoint {
```

Comment:
Add todo comment about encapsulating Endpoint into a class/class hierarchy. Currently ConnectionManager needs to know too much about the internals of Endpoints (buffer swithing, type casting etc)